datastream-common/src/test/java/com/linkedin/datastream/metrics/TestMetrics.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright 2019 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.metrics;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for {@link BrooklinMetrics}.
+ */
+public class TestMetrics {
+  @Test
+  public void testBrooklinMeterInfoComparison() {
```

Comment:
You can reduce the duplication between the identical tests in this file by using generics:
```java
  private static <T extends BrooklinMetricInfo> void testBrooklinMetricInfoHelper(
      Function<String, T> singleArgCtor, BiFunction<String, Optional<List<String>>, T> twoArgCtor) {

    T brooklinMeterInfo1 = singleArgCtor.apply ("meter1");
    T brooklinMeterInfo2 = singleArgCtor.apply("meter2");
    T brooklinMeterInfo3 = twoArgCtor.apply("meter1", Optional.of(Arrays.asList("hello", "world")));
    T brooklinMeterInfo4 = twoArgCtor.apply("meter1", Optional.of(Arrays.asList("world", "hello")));
    T brooklinMeterInfo5 = singleArgCtor.apply("meter1");

    Assert.assertNotEquals(brooklinMeterInfo1, brooklinMeterInfo2);
    Assert.assertNotEquals(brooklinMeterInfo1, brooklinMeterInfo3);
    Assert.assertEquals(brooklinMeterInfo3, brooklinMeterInfo4);
    Assert.assertEquals(brooklinMeterInfo1, brooklinMeterInfo5);
  }
```
which can then be used as:
```java
  @Test
  public void testBrooklinMetricInfo() {
    testBrooklinMetricInfoHelper(BrooklinMeterInfo::new, BrooklinMeterInfo::new);
    testBrooklinMetricInfoHelper(BrooklinCounterInfo::new, BrooklinCounterInfo::new);
    testBrooklinMetricInfoHelper(BrooklinGaugeInfo::new, BrooklinGaugeInfo::new);
    testBrooklinMetricInfoHelper(BrooklinHistogramInfo::new, BrooklinHistogramInfo::new);
  }
```